central/cluster/datastore/datastore_impl.go

```diff
@@ -420,10 +422,17 @@ func (ds *datastoreImpl) UpdateClusterHealth(ctx context.Context, id string, clu
 	if err != nil {
 		return err
 	}
+
 	if !exists {
 		return nil
 	}
 	cluster.HealthStatus = clusterHealthStatus
+
+	if oldHealth.GetSensorHealthStatus() == storage.ClusterHealthStatus_UNINITIALIZED &&
```

Comment:
Do I understand correctly that this is called whenever the cluster health changes? For example when one of the secured cluster components (sensor, collector, admission controller) changes health status. If so, `ClusterInitialized` doesn't seem like the best name because it may be called many times.

Reply:
Hence the filtering for the sensor state shift from UNINITIALIZED to something. Should normally pass once, as a sensor shouldn't get back to UNINITIALIZED.